huaweicloud/services/as/resource_huaweicloud_as_configuration.go

```diff
@@ -287,51 +295,44 @@ func getPublicIps(publicIpMeta map[string]interface{}) configurations.PublicIpOp
 	return publicIpOpts
 }
 
-func getInstanceConfig(configDataMap map[string]interface{}) (configurations.InstanceConfigOpts, error) {
+func buildInstanceConfig(configDataMap map[string]interface{}) (configurations.InstanceConfigOpts, error) {
 	disksData := configDataMap["disk"].([]interface{})
-	disks, err := getDisk(disksData)
+	disks, err := buildDiskOpts(disksData)
 	if err != nil {
-		return configurations.InstanceConfigOpts{}, fmtp.Errorf("Error happened when validating disk size: %s", err)
+		return configurations.InstanceConfigOpts{}, fmt.Errorf("the disk size is invalid: %s", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("the disk size is invalid: %s", err)
```

Reply:
the return value is not a pointer, I think we can keep it as is.